src/lib/cache/SubscriptionCache.cpp

```diff
@@ -0,0 +1,1000 @@
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include <semaphore.h>
+#include <regex.h>
+#include <stdint.h>
+#include <string>
+#include <vector>
+#include <time.h>
+
+#include "logMsg/logMsg.h"
+
+#include "common/statistics.h"
+#include "common/clockFunctions.h"
+
+#include "ngsi/Restriction.h"
+#include "ngsi/Reference.h"
+#include "ngsi/NotifyConditionVector.h"
+#include "ngsi/EntityId.h"
+#include "ngsi10/SubscribeContextRequest.h"
+#include "ngsi10/UpdateContextSubscriptionRequest.h"
+#include "mongoBackend/MongoGlobal.h"
+#include "cache/SubscriptionCache.h"
+
+
+
+namespace orion
+{
+
+
+/* ****************************************************************************
+*
+* subCache - global singleton for the subscription cache
+*/
+SubscriptionCache* subCache = NULL;
+
+
+
+/* ****************************************************************************
+*
+* subCacheMutexTime - 
+*/
+static struct timespec subCacheMutexTime = { 0, 0 };
+
+
+
+/* ****************************************************************************
+*
+* subscriptionCacheInit - 
+*/
+void subscriptionCacheInit(std::string dbName)
```

Comment:
Maybe it would be better to have the SubcriptionCache and EntityInfo classes separated in different files? I mean, EntityInfo.cpp/h and SubscriptionCache.cpp/h files for the EntityInfo::XXX and SubscriptionCache::XXX methods, plus an additional file for the global functions related with cache management such as subscriptionCacheInit.